chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -741,10 +757,7 @@ abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
       clone match {
         case Some(clone) =>
           clone._outerInst = this._outerInst
-          if (!clone.typeEquivalent(this)) {
-            autoClonetypeError(s"automatically cloned $clone not type-equivalent to base." +
-            " Constructor argument values were not inferred, ensure constructor is deterministic.")
-          }
+          validateClone(clone, "Constructor argument values were not inferred, ensure constructor is deterministic.")
```

Comment:
Why not run this on creation of `elements`? That will always happen on or before cloning

Reply:
@jackkoenig does #1050 address this concern?